xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/store/hibernate/HibernateAdapterFactory.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.store.hibernate;
+
+import java.sql.DatabaseMetaData;
+import java.util.Optional;
+
+import org.hibernate.HibernateException;
+import org.hibernate.cfg.Configuration;
+import org.xwiki.component.annotation.Role;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Factory in charge of providing a {@link HibernateAdapter} instance.
+ * 
+ * @version $Id$
+ * @since 17.1.0RC1
+ */
+@Unstable
+@Role
+public interface HibernateAdapterFactory
+{
+    /**
+     * Create a new {@link HibernateAdapter} instance corresponding to the passed {@link DatabaseMetaData}.
```

Comment:
it gets more than it creates since they're all singleton

Reply:
That's an implementation detail of the current ones, some might not be singletons.